### PR TITLE
Add cluster resize

### PR DESCRIPTION
**Pull request by mesellings**
## Description

Closes https://github.com/camunda/developer-experience/issues/409.

Add content for how to resize a cluster as part of the new update cluster size feature.

**Note:** This will need to be backported for 8.6 once initial Engineering review is complete.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [X] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [X] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [X] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/best-practices/architecture/sizing-your-environment.md [hunk 1]
```diff
@@ -146,7 +146,7 @@ Now you can select a hardware package that can cover these requirements. In this
 Camunda 8 defines four [cluster sizes](/components/concepts/clusters.md#cluster-size) you can select from (1x, 2x, 3x, and 4x) after you have chosen your [cluster type](/components/concepts/clusters.md#cluster-type). The following table gives you an indication of what requirements you can fulfill with each cluster size.
 
 :::note
-Contact your Customer Success Manager if you require a custom cluster size above these requirements.
+Contact your Customer Success Manager to increase the cluster size beyond the maximum 4x size. This requires custom sizing and pricing.
```
Comment (ultraschuppi) [inline docs/components/best-practices/architecture/sizing-your-environment.md:149]: the issue i am seeing here is that only new or renegiotiated contracts will actually get this - as the advanced-offering packages need to be unlocked for them first...

File: docs/components/concepts/clusters.md [hunk 2]
```diff
@@ -46,22 +46,18 @@ See [Camunda Enterprise General Terms](https://legal.camunda.com/licensing-and-o
 
 The cluster size defines the cluster performance and capacity.
 
-After you have chosen your cluster type, you can choose the cluster size that best meets your cluster environment requirements.
+After you have chosen your cluster type, choose the cluster size that best meets your cluster environment requirements.
```
Comment (ultraschuppi) [inline docs/components/concepts/clusters.md:49]: same for this one

File: docs/components/best-practices/architecture/sizing-your-environment.md (hunk 1 above)
Comment (mesellings) [reply docs/components/best-practices/architecture/sizing-your-environment.md:149]: @ultraschuppi I have added some text to the note to clarify this a bit more: 
image

**Comment by mesellings** [comment]
@ultraschuppi @felix-mueller I've backported the changes to the 8.6 docs, changed the link to match the product (Resize cluster), and a few more edits for conciseness - please re-review 👍

**Review by christinaausley — APPROVED** [review]
Chefs kiss 👨‍🍳 🤌

**Pull request merged**